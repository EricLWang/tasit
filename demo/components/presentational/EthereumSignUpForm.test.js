import React from "react";
import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import { shallow } from "enzyme";
import EthereumSignUpForm from "./EthereumSignUpForm";

describe("EthereumSignUpForm", () => {
  jest.useFakeTimers();
  beforeEach(() => {
    NavigationTestUtils.resetInternalState();
  });

  it("renders the component", async () => {
    expect(shallow(<EthereumSignUpForm />)).toMatchSnapshot();
  });

  it("creates a wallet - calling function", async () => {
    const wrapper = shallow(<EthereumSignUpForm />);
    expect(wrapper.state("address")).toEqual("");
    await wrapper.instance().createAccount();
    expect(wrapper.state("address")).not.toEqual("");
  });
});
