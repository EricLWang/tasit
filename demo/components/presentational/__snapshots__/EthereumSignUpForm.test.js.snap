// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EthereumSignUpForm renders the component 1`] = `
<Fragment>
  <Component
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Component
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
        }
      }
    >
      <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoCorrect={false}
        onChangeText={[Function]}
        placeholder="username"
        style={
          Object {
            "fontSize": 20,
            "justifyContent": "flex-start",
            "width": 90,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </Component>
    <Component
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <Component
        style={
          Object {
            "fontSize": 20,
            "justifyContent": "flex-end",
          }
        }
      >
        .tasitid.eth
      </Component>
    </Component>
  </Component>
  <Component
    style={
      Object {
        "flexDirection": "row",
        "marginTop": 30,
      }
    }
  >
    <Button
      onPress={[Function]}
      title="Continue"
    />
  </Component>
</Fragment>
`;
